File: projects/faad2/fuzz_config.c
/* Copyright 2021 Google LLC
Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at
      http://www.apache.org/licenses/LICENSE-2.0
Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/
#include <stddef.h>
#include <stdint.h>
#include <string.h>
#include <stdlib.h>

#include "neaacdec.h"

int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
  unsigned char* non_const_data = (unsigned char *)malloc(size);
  memcpy(non_const_data, data, size);
  mp4AudioSpecificConfig mp4ASC;

  NeAACDecAudioSpecificConfig(non_const_data, (unsigned long) size, &mp4ASC);
  free(non_const_data);

  return 0;
}

File: projects/g-api-py-oauthlib/fuzz_config.py
#!/usr/bin/python3
# Copyright 2022 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#      http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

import os
import sys
import mock
import atheris
import tempfile

import json
import mock
from google_auth_oauthlib import helpers


def TestOneInput(data):
    fdp = atheris.FuzzedDataProvider(data)
    config_file = tempfile.NamedTemporaryFile(mode="wt")
    try:
        payload = json.loads(fdp.ConsumeUnicodeNoSurrogates(1024))
    except:
        return
    if type(payload) is not dict:
        return
    config_file.write(json.dumps(payload))

    try:
        helpers.session_from_client_secrets_file(
            config_file.name, scopes=mock.sentinel.scopes
        )
    except ValueError as ve:
        legit_exceptions = [
            "Client secrets must be for a web or installed app.",
            "Client secrets is not in the correct format."
        ]
        legit = False
        for msg in legit_exceptions:
            if msg in str(ve):
                legit = True
        if not legit:
            raise ve


def main():
    atheris.instrument_all()
    atheris.Setup(sys.argv, TestOneInput, enable_python_coverage=True)
    atheris.Fuzz()


if __name__ == "__main__":
    main()
